src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -1333,6 +1334,29 @@ ActionsDAGPtr SelectQueryExpressionAnalyzer::appendOrderBy(ExpressionActionsChai
             with_fill = true;
     }
 
+    if (auto interpolate_list = select_query->interpolate())
+    {
+        auto find_columns = [&step](IAST * function)
+        {
+            auto f_impl = [&step](IAST * fn, auto fi)
+            {
+                if (auto * ident = fn->as<ASTIdentifier>())
+                {
+                    step.addRequiredOutput(ident->getColumnName());
```

Comment:
It looks like here we add alias to required output instead of full column name.
At list this does not work:
```
SELECT
    n,
    source,
    inter + 1 AS inter_p
FROM
(
    SELECT
        toFloat32(number % 10) AS n,
        'original' AS source,
        number AS inter
    FROM numbers(10)
    WHERE (number % 3) = 1
)
ORDER BY n ASC WITH FILL FROM 0 TO 11.51 STEP 0.5
INTERPOLATE ( inter_p AS inter_p + 1 )

Query id: 9c5b264e-0dca-406e-a2c3-7d9dd43f6609


0 rows in set. Elapsed: 0.002 sec. 

Received exception from server (version 22.4.1):
Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Missing columns: 'inter_p' while processing query: 'SELECT n, source, inter + 1 AS inter_p FROM (SELECT toFloat32(number % 10) AS n, 'original' AS source, number AS inter FROM numbers(10) WHERE (number % 3) = 1) ORDER BY n ASC WITH FILL FROM 0 TO 11.51 STEP 0.5 INTERPOLATE ( inter_p AS inter_p + 1 )', required columns: 'n' 'inter_p' 'source' 'inter' 'n' 'inter_p' 'source' 'inter'. (UNKNOWN_IDENTIFIER)
```

Reply:
Maybe we can build actions for `interpolate` here as well?

Now I think that maybe it would be easier to substitute aliases for interpolate.
Like, fix QueryNormalizer a little bit (add ASTInterpolateElement)
https://github.com/ClickHouse/ClickHouse/blob/5a1392a8e349c2bba7e35b01b3e4f2b3c095aeae/src/Interpreters/QueryNormalizer.cpp#L251-L262
so that in AST `inter_p AS inter_p + 1` would be rewritten into `inter_p AS (inter + 1) + 1`

Reply:
it's rather I shouldn't add this column as required because it's not a source column...